### PR TITLE
initial commit for tracing in ambassador

### PR DESCRIPTION
**Pull request by RoryShively**
Here's my initial commit on tracing in ambassador

### CODE REVIEW

**Comment by richarddli** [comment]
Thanks! See issue #339

File: ambassador/ambassador/config.py [hunk 1]
```diff
@@ -1313,6 +1317,40 @@ def tls_config_helper(self, name, amod, tmod):
...
+        tmp_config["collector_cluster"] = collector_cluster
+
+        for key in tracing_config.keys():
+            if key.startswith('_'):
+                continue
+
+            if key == 'driver':
+                driver = tracing_config.get(key, '').lower()
+                # Only allow user to create a 'zipkin' or 'lightstep' driver for now. Update with envoy
+                assert driver in ['zipkin', 'lightstep']
```
Comment (kflynn) [inline ambassador/ambassador/config.py]: We don't use `assert` for error handling anywhere else; better to use `post_error` instead. See `Config.add_clusters_for_mapping` and `Config.auth_helper` for examples.

File: ambassador/ambassador/config.py [hunk 2]
```diff
@@ -1324,11 +1362,14 @@ def module_config_ambassador(self, name, amod, tmod):
...
 
+        if amod and ('tracing' in amod):
+            self.tracing_config_helper(name, amod)
+
         # After that, check for port definitions, probes, etc., and copy them in
         # as we find them.
         for key in [ 'service_port', 'admin_port', 'diag_port',
                      'liveness_probe', 'readiness_probe', 'auth_enabled',
-                     'use_proxy_proto', 'use_remote_address' ]:
+                     'use_proxy_proto', 'use_remote_address', 'tracing', ]:
```
Comment (kflynn) [inline ambassador/ambassador/config.py]: I feel like this is redundant with the `self.tracing_config_helper` call above -- feels like you can just remove `tracing` from the array here?

**Comment by kflynn** [comment]
@RoryShively Thanks! A couple of comments inline -- can you handle the conflict and address those?

We need to spin a new `ambassador-envoy` 1.6.0 so I'm going to try to get that done quickly, to be able to land it before this one. Thanks again!

**Comment by RoryShively** [comment]
I'll try and get to this asap. I got pulled onto another project for the time being

**Comment by kflynn** [comment]
@RoryShively That'd be wonderful. :)

**Comment by alexgervais** [comment]
@RoryShively any progress on this? I'm really interested in tracing, I could help!

**Comment by richarddli** [comment]
Closing since this has been resolved in #655

**Pull request closed without merging**